infrastructure/code/convert.go

```diff
@@ -278,10 +279,11 @@ func (s *SarifResponse) toIssues() (issues []snyk.Issue) {
 			// convert the documentURI to a path according to our conversion
 			path := loc.PhysicalLocation.ArtifactLocation.URI
 
-			startLine := loc.PhysicalLocation.Region.StartLine - 1
-			endLine := loc.PhysicalLocation.Region.EndLine - 1
-			startCol := loc.PhysicalLocation.Region.StartColumn - 1
-			endCol := loc.PhysicalLocation.Region.EndColumn
+			position := loc.PhysicalLocation.Region
+			startLine := position.StartLine - 1
+			endLine := util.Max(position.EndLine-1, startLine)
+			startCol := position.StartColumn - 1
+			endCol := util.Max(position.EndColumn-1, startCol)
```

Comment:
Surprising that the standard lib doesn't provide a generified max. Anyhoo, love it.

Reply:
`endColumn` is not correct I think. If you have a multiline finding, the end column can be smaller than the start column.

Reply:
Good call on the multiline case, I'll set the 2nd parameter to `0`.

In my testing it looked like the EndColumn was always off by 1, but we can verify.

Reply:
And regarding the max function - agree. [Some languages](https://learn.microsoft.com/en-us/dotnet/api/system.math.max?view=netframework-1.1) supported this back in 2002 (thanks to function overloading), way before [generic math](https://learn.microsoft.com/en-us/dotnet/standard/generics/math) was a thing.